hs_core/task_utils.py

```diff
@@ -28,15 +37,196 @@ def _retrieve_task_id(res_id, job_dict):
     return None
 
 
+def _retrieve_user_tasks(username, job_dict, queue_type):
```

Comment:
Wondering if you should omit the queue_type and return all 3 types with one call so we walk all workers/tasks only once instead of 3 times.

Reply:
nvm, it's walking different job_dicts each time